sdk/python/tests/integration/e2e/test_universal_e2e.py

```diff
@@ -3,27 +3,45 @@
 from typing import Optional
 
 import pandas as pd
+import pytest
 from pytz import utc
 
 from feast import FeatureStore, FeatureView
-from tests.integration.feature_repos.test_repo_configuration import (
-    Environment,
-    parametrize_e2e_test,
+from tests.data.data_creator import create_dataset
+from tests.integration.feature_repos.repo_configuration import (
+    FULL_REPO_CONFIGS,
+    construct_test_environment,
+    vary_infer_feature,
 )
 from tests.integration.feature_repos.universal.entities import driver
 from tests.integration.feature_repos.universal.feature_views import driver_feature_view
 
 
-@parametrize_e2e_test
-def test_e2e_consistency(test_environment: Environment):
-    fs, fv = (
-        test_environment.feature_store,
-        driver_feature_view(test_environment.data_source),
-    )
-    entity = driver()
-    fs.apply([fv, entity])
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "config", vary_infer_feature(FULL_REPO_CONFIGS), ids=lambda v: str(v)
+)
+def test_e2e_consistency(config):
+    df = create_dataset()
+    test_suite_name = test_e2e_consistency.__name__
+    with construct_test_environment(config, test_suite_name) as test_environment:
+        fs = test_environment.feature_store
+        infer_features = test_environment.test_repo_config.infer_features
+        _, _, data_source, _ = (
+            None,
+            df,
+            test_environment.data_source_creator.create_data_source(
+                df, fs.project, field_mapping={"ts_1": "ts", "id": "driver_id"}
+            ),
+            test_environment.data_source_creator,
+        )
```

Comment:
It's a little bit unclear what is happening here. Why do we create a tuple if we can create the `data_source` directly?

Reply:
Clean up